yt_dlp/YoutubeDL.py

```diff
@@ -1630,18 +1631,29 @@ def progress(msg):
                 self.to_screen('')
             raise
 
-    def _load_cookies(self, data, *, from_headers=True):
+    def _remove_cookie_header(self, http_headers):
+        """Filters out `Cookie` header from an `http_headers` dict
+
+        The `Cookie` header is removed to prevent leaks and supercookies.
+        See: https://github.com/yt-dlp/yt-dlp/security/advisories/GHSA-v8mc-9377-rwjj
+
+        @param http_headers     An `http_headers` dict with a `Cookie` header to be removed
+        """
+        return dict(traverse_obj(http_headers, ({dict.items}, lambda _, pair: pair[0].lower() != 'cookie')))
```

Comment:
This only needs traversal if `http_headers` may not be `dict`-like. If that can be assumed, `filter(lambda pair: pair[0].lower() != 'cookie', http_headers.items())` is enough.

Reply:
`info_dict.get('http_headers')` is passed to the method here when it can return `None`:
https://github.com/yt-dlp/yt-dlp/pull/7546/files#diff-d3ba8be45cae8dd7889a71c3360c9e4ac1160de8a5f3443b6e4a656395267f9bR2777

Although now I'm not sure the linked line of code is necessary, since `test_infojson_cookies` passes without it

Reply:
This will eventually be obsoleted by `CaseInsensitiveDict` anyways